packages/runtime/src/runtime/hooks/useEnsembleAction.tsx

```diff
@@ -120,6 +131,163 @@ export const useInvokeApi: EnsembleActionHook<InvokeAPIAction> = (action) => {
   return invokeApi;
 };
 
+export const usePickFiles: EnsembleActionHook<PickFilesAction> = (
+  action?: PickFilesAction,
+) => {
+  const [files, setFiles] = useState<FileList>();
+  const onCompleteAction = useEnsembleAction(action?.onComplete);
+
+  useRegisterBindings(
+    {
+      files,
+    },
+    action?.id,
+    {
+      setFiles,
+    },
+  );
+
+  const inputEl = useMemo(() => {
+    const input = document.createElement("input");
+    input.type = "file";
+    input.multiple = action?.allowMultiple || false;
+    input.accept =
+      action?.allowedExtensions?.map((ext) => ".".concat(ext))?.toString() ||
+      "*/*";
+
+    input.onchange = (event: Event): void => {
+      const selectedFiles =
+        (event.target as HTMLInputElement).files || undefined;
+
+      if (selectedFiles) setFiles(selectedFiles);
+
+      if (onCompleteAction) onCompleteAction?.callback();
```

Comment:
This should be nested inside selectedFiles. I don't believe the the onComplete action should fire if no files were successfully picked?